### ISSUE TITLE
[Bug]: src prop has the wrong type

### ISSUE THREAD

**Issue by hydro63** [Elius94/react-photo-sphere-viewer#68]
### What happened?

I was trying to get the [Cubemap](https://photo-sphere-viewer.js.org/guide/adapters/cubemap.html#example) adapter to work with the component, but TS was protesting passing the configuration through the `src`. TS expects the `src` to be `string`, despite the core library allowing also `array` and `object`.

*\*PS: this caused me to try to use the `panorama` prop, which caused even more problems*

### What should have happened?

TS should not have raised an error for a valid input.

### Solution
I propose allowing the `src` prop to be of type `string | string[] | object` and depracating the `panorama` prop.

### Code

```tsx

// TS raises error with src (expected string)
<ReactPhotoSphereViewer
  src={{
    left: './cubemap/_nx.jpg',
    front: './cubemap/_pz.jpg',
    right: './cubemap/_px.jpg',
    back: './cubemap/_nz.jpg',
    top: './cubemap/_py.jpg',
    bottom: './cubemap/_ny.jpg',
  }}
  height='80svh'
  width='60%'
  adapter={CubemapAdapter}
></ReactPhotoSphereViewer>
```
  

### Sandbox Link

https://codesandbox.io/p/sandbox/htqxyv

### Library Version

6.0.0

### What operating system are you using?

Windows

### What browser are you using?

Chrome

### Logs

couldn't get it to log anything

### Interest to fix the bug

- [ ] I would like to fix this bug!

**Comment by Elius94** [comment]
Hi, The sandbox url is not working, please fix

**Comment by hydro63** [comment]
Sorry, forgot to set it as public. changed the link, it should be working now.

LINK: https://codesandbox.io/p/sandbox/htqxyv

**Comment by hydro63** [comment]
@Elius94 After more googling, and reading the docs of the core library thoroughly, i've found out i made a wrong assumption, which caused some additional problems. I've repurposed and edited this issue to talk about the real bug which allowed some of the bad things to happen - the `src` overly specific type.

**Comment by Elius94** [comment]
Ok, so, now you have another problem?

**Comment by hydro63** [comment]
Not exactly, it's the same problem, it's just that i've more specifically identified the exact cause.

PS: i'm really sorry for the inconvenience

**Comment by Elius94** [comment]
Fixed in the new version :)
